src/test/java/com/synopsys/integration/jenkins/detect/service/DetectArgumentServiceTest.java

```diff
@@ -0,0 +1,210 @@
+
+    private final IntEnvironmentVariables intEnvironmentVariables = new IntEnvironmentVariables();
+    private final TaskListener taskListener = Mockito.mock(TaskListener.class);
+    private final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+    private final JenkinsIntLogger jenkinsIntLogger = new JenkinsIntLogger(taskListener);
+
+    private final DetectJarStrategy detectJarStrategy = new DetectJarStrategy(jenkinsIntLogger, intEnvironmentVariables, expectedJavaHome, expectedDetectPath);
+    private final DetectArgumentService detectArgumentService = new DetectArgumentService(jenkinsIntLogger, jenkinsVersionHelper);
+
+    private Map<String, String> inputDetectProperties = new LinkedHashMap<>();
+    private Map<String, String> expectedVisibleDetectProperties = new LinkedHashMap<>();
+    private Map<String, String> expectedHiddenDetectProperties = new LinkedHashMap<>();
+
+    @BeforeEach
+    public void setUp() {
+        jenkinsIntLogger.setLogLevel(LogLevel.DEBUG);
+        intEnvironmentVariables.put("DETECT_PLUGIN_ESCAPING", "true");
+
+        // Set default expected values. Tests will apply different if needed
+        Mockito.when(taskListener.getLogger()).thenReturn(new PrintStream(byteArrayOutputStream));
+        Mockito.when(jenkinsVersionHelper.getJenkinsVersion()).thenReturn(Optional.of(expectedJenkinsVersion));
+        Mockito.when(jenkinsVersionHelper.getPluginVersion("blackduck-detect")).thenReturn(Optional.of(expectedJenkinsPluginVersion));
+
+        // These are the standard input detect properties
+        inputDetectProperties.put("--detect.docker.passthrough.service.timeout", "1234567890");
+        inputDetectProperties.put("--detect.cleanup", "ThisIsFalse");
+
+        // These are the standard expected detect properties. Tests will update if needed.
+        // They SHOULD be visible in the output log message and be within the return List
+        expectedVisibleDetectProperties.putAll(inputDetectProperties);
+        expectedVisibleDetectProperties.put(loggingLevelKey, jenkinsIntLogger.getLogLevel().toString());
+
+        // These are the automatically appended detect properties added by DetectArgumentService().
+        // They should NOT be visible in the output log message but should be within the return List
+        expectedHiddenDetectProperties.put(jenkinsVersionParam, expectedJenkinsVersion);
+        expectedHiddenDetectProperties.put(pluginVersionParam, expectedJenkinsPluginVersion);
+
+        DetectJarStrategy detectJarStrategy = new DetectJarStrategy(jenkinsIntLogger, intEnvironmentVariables, expectedJavaHome, expectedDetectPath);
+        DetectArgumentService detectArgumentService = new DetectArgumentService(jenkinsIntLogger, jenkinsVersionHelper);
+    }
+
+    private String createDetectPropertiesInputString() {
+        return inputDetectProperties.keySet().stream()
+                   .map(key -> key + "=" + inputDetectProperties.get(key) + " ")
```

Comment:
This will add an extra space at the end of the properties string-- it likely doesn't matter much, but it's also not likely what you want. You can do Collectors.joining(" ") which will give you what you want.